cli/src/lockfiles/mod.rs

```diff
@@ -22,6 +22,13 @@ pub type ParseResult = anyhow::Result<Vec<PackageDescriptor>>;
 
 pub trait Parseable {
     fn new(filename: &Path) -> Result<Self, io::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Self::from_string(std::fs::read_to_string(filename)?))
+    }
```

Comment:
Is the `new()` function still called anywhere?

Reply:
Only tests. I've considered removing it, but then you have a bunch of tests that all have to manually do this. So I thought I might as well keep it. I'd probably rename it though, to not favor it over `from_string`?